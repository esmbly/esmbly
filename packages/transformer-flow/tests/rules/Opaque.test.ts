import testRunner from '../helpers/testRunner';

const input = `
  /* @flow */

  opaque type Opaque = number;
`;

const expected = `
  type Opaque = number;
`;

describe('rule: Opaque', () => {
  it('correctly transforms the input to TypeScript', async () => {
    const [{ content }] = await testRunner(input);
    expect(content).toEqual(expected);
  });
});
