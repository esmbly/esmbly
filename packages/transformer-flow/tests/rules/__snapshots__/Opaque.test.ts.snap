// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule: Opaque produces a correctly formated warning 1`] = `
"
  [33mwarning[39m [2mOpaque types can't be expressed in TypeScript.[22m

    [31mopaque type Opaque = number;[39m
  
  [2mat:[22m [36minput.js:4:2[39m
  [2missue:[22m [36mhttps://github.com/Microsoft/TypeScript/issues/202[39m

"
`;
